src/codemagic/cli/argument/argument_parser_builder.py

```diff
@@ -123,16 +133,40 @@ def _get_custom_argument_group(self, group_name) -> ArgumentGroup:
             self._custom_arguments_groups[group_name] = argument_group
         return argument_group
 
+    def _get_custom_mutually_exclusive_required_group(self, group_name: str) -> ArgumentGroup:
+        try:
+            argument_group = self._custom_arguments_groups[group_name]
+        except KeyError:
+            group_description = Colors.UNDERLINE(
+                f"One of following arguments are required for "
+                f"{Colors.BOLD(self._cli_action.action_name)} to {Colors.BOLD(group_name)}",
+            )
+            group = self._action_parser.add_argument_group(group_description)
+            argument_group = group.add_mutually_exclusive_group(required=True)
+            self._custom_arguments_groups[group_name] = argument_group
+        return argument_group
+
     def _get_argument_group(self, argument) -> ArgumentGroup:
-        if argument.argument_group_name is None:
+        if argument.argument_group_name is None and argument.mutually_exclusive_required_group is None:
```

Comment:
If this if-else was somewhat reversed and carried out like
```python
if argument.argument_group_name:
    argument_group = self._get_custom_argument_group(...)
elif argument.mutually_exclusive_required_group:
    argument_group = self._get_custom_mutually_exclusive_required_group(...)
elif argument.is_required():
    argument_group = self._required_arguments
else:
    argument_group = self._optional_arguments
```
Like that we could also get rid of the intermediate `_get_custom_argument_group` method.